docs/website/contents/for-developers/HardForks.md

```diff
@@ -0,0 +1,98 @@
+
+There are other Cardano networks besides mainnet, notably the testnets.
+By their essential nature, though, testnets should also use mainnet's mapping between each major protocol version number and the corresponding pair of protocol rules and ledger rules.
+On the other hand, these other networks must be able to hard fork at different times than mainnet does --- the testnets should have hard forked successfully before mainnet hard forks!
+Since the testnet governance and mainnet governance are independent, the determination of the hard fork timing via on-chain governance immediately enables it to vary on different networks' chains.
+
+## Eras
+
+A hard fork is the result of the nodes' developers introducing backwards-incompatible changes [^change-decisions] and the community ultimately choosing to adopt them by increasing the major protocol version parameter via on-chain governance [^frame-rule].
+When those developers design and implement such changes, they determine whether the changes are easy to maintain as variations within the existing code.
+If so, then the hard fork is ultimately toggling some conditionals within the code.
+If the changes are not so simple, the Cardano developers instead introduce a new _era_ within the code [^roadmap-disambiguation].
+
+Each era defines some new types and rules that are independent within the code from the existing eras.
+For example, the [Conway era introduces](https://docs.cardano.org/about-cardano/evolution/upgrades/chang) many new parts of the ledger state, transaction types, corresponding rules, etc that the preceding era does not have, and so the architects decided to explicitly separate the Conway code from the old code as a new era.
+The Ledger and Consensus infrastructure code is defined so that each era can flexibly reuse whichever of the older eras' features it does not supersede.
+Several eras coexist sequentially on the Cardano chain; a hard fork that takes one step within that sequence is called an _era transition_.
+
+Every hard fork is either an era transition or an _intra-era hard fork_ [^accidentals].
+Intra-era hard forks only introduce changes with a small effect on the code, eg a few easy-to-maintain conditionals.
+Some have been bugfixes that were small by chance, and others have been changes that were only small because they had been anticipated and intentionally deferred when the era was originally designed.
+
+Eras are a straight-forward mechanism that enables node developers to separate the concerns arising from features that are added to Cardano over time.
+The separation amongst eras inherently simplifies the development of the new features, and it also helps ensure that the Cardano node can sync the entire historical chain [^mithril] --- as well as refuse invalid alternative histories.
+
+### Which changes require a new era?
```

Comment:
Another reason for a new era is that sometimes it happens that new features that are being introduced require an initial value, which needs to be populated upon a hard fork. For example initial Constitution and the guardrails script in Conway. Such initial values are supplied through a Genesis file.